### PR TITLE
Enhancement/676

### PR DESCRIPTION
**Pull request by shintung00**
UI
Create new component GenerateModelControl.tsx to determine when scene generation should be allowed
GenerateModelControl also has explicit instructions on how to enable scene generation during model ingestion

Client/Util
Created a helper method enableSceneGenerateCheck to determine when enable scene generation control can be enabled during model ingestion

Workflow
Added a new flag to the eventIngestionIngestObjectModel params that indicates whether to skip scene generation

### CODE REVIEW

File: client/src/pages/Ingestion/components/Metadata/Control/GenerateModelControl.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
```
Comment (jahjedtieson) [inline client/src/pages/Ingestion/components/Metadata/Control/GenerateModelControl.tsx:1]: I'll be honest here -- I don't love the name of this control.  It sounds like something used to control generation of models.  While it's true that scene generation produces models, this UI is really about controlling scene generation. Please consider a rename.

File: client/src/store/metadata/metadata.defaults.ts [hunk 2]
```diff
@@ -175,7 +175,8 @@ export const defaultModelFields: ModelFields = {
         selected: true,
         subtitleOption: eSubtitleOption.eNone,
         id: 0
-    }]
+    }],
+    skipSceneGenerate: true
```
Comment (jahjedtieson) [inline client/src/store/metadata/metadata.defaults.ts:179]: I'm not sure about the impact of this default being set to `true`, but the default behavior is for us to generate scenes, not skip scene generation.  Perhaps this should be updated to `false`?

File: client/src/store/utils.ts [hunk 3]
```diff
@@ -153,3 +155,27 @@ export function parseSubtitlesToState(titles: IngestTitle): SubtitleFields {
 
     return result;
 }
+
+export function enableSceneGenerateCheck(metadata: ModelFields, unitsEntries: VocabularyOption[], purposeEntries: VocabularyOption[], typeEntries: VocabularyOption[]): boolean {
+    const { units, purpose, modelFileType } = metadata;
+    if (!units || !purpose || !modelFileType) return false;
+    const eVocabUnitSet = new Set([eVocabularyID.eModelUnitsMillimeter as number, eVocabularyID.eModelUnitsCentimeter, eVocabularyID.eModelUnitsMeter, eVocabularyID.eModelUnitsInch, eVocabularyID.eModelUnitsFoot, eVocabularyID.eModelUnitsYard]);
```
Comment (jahjedtieson) [inline client/src/store/utils.ts:162]: Given that `eVocabUnitSet`, `eVocabPurposeSet`, and `eVocabFileTypeSet` are const and never change, perhaps these should be defined at the module level and not inside the function, to ensure they are evaluated just once, and not every time this method is called.

I'm not sure how javascript optimization works here -- perhaps function-scoped const items are evaluated just once. But I suspect they are evaluated each time the method executes.

File: client/src/pages/Ingestion/components/Metadata/Control/GenerateModelControl.tsx (hunk 1 above)
Comment (shintung00) [reply client/src/pages/Ingestion/components/Metadata/Control/GenerateModelControl.tsx:1]: How does SceneGenerateWorkflowControl sound?

File: server/graphql/schema/ingestion/resolvers/mutations/ingestData.ts [hunk 4]
```diff
@@ -1359,14 +1364,15 @@ class IngestDataWorker extends ResolverBase {
                     }
                 }
             }
+            if (typeof AVInfo.skipSceneGenerate === 'boolean') IAR.skipSceneGenerate = AVInfo.skipSceneGenerate;
```
Comment (jahjedtieson) [inline server/graphql/schema/ingestion/resolvers/mutations/ingestData.ts:1367]: What's the purpose of the test `if (typeof AVInfo.skipSceneGenerate === 'boolean')` here? Why not just assign `IAR.skipSceneGenerate = AVInfo.skipSceneGenerate;`?

Comment (shintung00) [reply server/graphql/schema/ingestion/resolvers/mutations/ingestData.ts:1367]: My original reasoning was that it's still possible for AVInfo to have null/undefined so this was added to handle those cases.
But thinking about it, it does seem a bit overly specific. I'll make the revision

**Review by jahjedtieson — APPROVED** [review]

**Pull request merged**